apps/settings/js/panels/improveBrowserOS/panel.js

```diff
@@ -0,0 +1,34 @@
+define(function(require) {
+  'use strict';
+
+  var SettingsPanel = require('modules/settings_panel');
+
+  return function ctor_improveBrowserOS() {
+    return SettingsPanel({
+      onInit: function(panel) {
+        this.openAppStats = panel.querySelector('#openAppStats');
+        this.openAppStats.addEventListener('click',
+                                          this._onOpenAppStats.bind(this));
+      },
+
+      onUninit: function() {},
+      onShow: function() {},
+      onHide: function() {},
+
+      _onOpenAppStats: function() {
+        var activity = new window.MozActivity({
+          name: 'marketplace-search',
+          data: {
+            type: 'firefox-os-app-stats'
+          }
+        });
+
+        activity.onerror = function() {
+          window.open(
```

Comment:
This way when users cancel the activity request they are still prompted with the dialog. We should not do this when `activity.error.name === 'ActivityCanceled'`.